### ISSUE TITLE
Help for transfer learning on a custom dataset

### ISSUE THREAD

**Issue by alient12** [ultralytics/ultralytics#4463]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Thanks for this great repo. I have some questions, the questions are about **Dataset for YOLO** but my results are from YOLOv7. I really appreciate for any help and recommendation from anyone.

This is what I did and my questions are at the end.
# Dataset
I have 2 datasets for fresh and rotten apple, banana, orange, let's say dataset A and B.

**Dataset A:**
- Has 1200 images, 200 for each class.
- All the images are captured by camera
- High resolution => resized to 640x640
- Is not augmented which is recommended for YOLO
- I divided it to 0.8 train, 0.1 valid, 0.1 test
- No bounding box label =>I labeled it myself in YOLO format
- Most images have light background

**Dataset B:**
- Most pictures are from internet and are unrealistic with paper white background
- The dataset is augmented
- Low resolution ~250-450
- 11000 train (each class ~1500) and 3000 test
- No bounding box label

# Training
I used transfer learning on Dataset A with YOLOv7 tiny. Changed number of classes to 83 and mapped fresh apple, fresh orange and fresh banana with their index in COCO and added 3 new indexes for rotten apple, rotten banana and rotten orange. This is the result after 300 epochs:

**validation:**
| P | R | mAP@0.5 | mAP@.5:.95 |
| :-- | :-- | :--| :--|
| 0.9706 | 0.9438 | 0.9826 | 0.9005 |

**test**
| P | R | mAP@0.5 | mAP@.5:.95 |
| :-- | :-- | :--| :--|
| 0.96 | 0.956 | 0.981 | 0.887 |

![results](https://github.com/WongKinYiu/yolov7/assets/73688480/76094cae-d514-44c3-ac69-ece067b8a36a)

The results say it's ok but when I tested it on Dataset B it wasn't ok.
**some tested images from Dataset B:**

| ![rotated_by_15_Screen Shot 2018-06-08 at 5 14 20 PM](https://github.com/WongKinYiu/yolov7/assets/73688480/6e44e963-d4ec-429f-8ae0-55bc2b265e39) | ![rotated_by_15_Screen Shot 2018-06-07 at 2 25 26 PM](https://github.com/WongKinYiu/yolov7/assets/73688480/2081c990-7a66-4fcd-b541-b6ee6f02ac14) | ![rotated_by_75_Screen Shot 2018-06-08 at 2 34 47 PM](https://github.com/WongKinYiu/yolov7/assets/73688480/786ded8e-3258-4bae-ab63-51380c201cba) |
| :-- | :-- | :--|

# Questions
1. Did my model overfit because of 300 epochs are too high and 1200 images are too low?
2. Is it fair to test model trained on 640x640 images on a lower size images?
3. How can I combine Dataset B with Dataset A since B is low quality (augmented, lower resolution)?
4. My last question is about labeling. In some images there are lots of apples (around 20, a few of them are on surface and the others are behind), I tagged all of them. Does it increase False Positive? Should I tag them as one? I tagged them separately because I wanted to use it on a robot arm application.

| ![FreshApple (27)](https://github.com/WongKinYiu/yolov7/assets/73688480/387c1ccd-37a3-4b0f-bce6-e65b44a4280a) | ![image](https://github.com/WongKinYiu/yolov7/assets/73688480/c9e31761-e415-400b-ada6-ed1c3c64381f) | 
| :-- | :-- |
